packages/stats/gbstats/gbstats.py

```diff
@@ -258,6 +261,9 @@ def analyze_metric_df(
             df[f"v{i}_uplift"] = None
             df[f"v{i}_error_message"] = None
 
+    if analysis.bandit and analysis.decision_metric == metric.id:
+        df = get_bandit_weights(df, metric, analysis)
```

Comment:
Two notes:

1. Seems like we can do this within `analyze_row`. I'd consider just getting rid of `get_bandit_weights` that takes a `df` and instead just pass in a row (or rename `get_bandit_weights_single`) and then set the values right in the copied row.
2. This also avoids overwiting the `df` like this, which is a little scary.

Reply:
As discussed offline, I have now made `get_bandit_weights` standalone, and not nested inside of `analyze_row`.